session/bootstrap.go

```diff
@@ -2002,3 +2004,32 @@ func oldPasswordUpgrade(pass string) (string, error) {
 	newpass := fmt.Sprintf("*%X", hash2)
 	return newpass, nil
 }
+
+// rebuildAllPartitionValueMapAndSorted rebuilds all value map and sorted info for list column partitions with InfoSchema.
+func rebuildAllPartitionValueMapAndSorted(s *session) {
+	type partitionExpr interface {
+		PartitionExpr() (*tables.PartitionExpr, error)
+	}
+
+	p := parser.New()
+	is := s.GetInfoSchema().(infoschema.InfoSchema)
+	for _, dbInfo := range is.AllSchemas() {
+		for _, t := range is.SchemaTables(dbInfo.Name) {
+			pi := t.Meta().GetPartitionInfo()
+			if pi == nil || pi.Type != model.PartitionTypeList {
+				continue
+			}
+
+			pe, err := t.(partitionExpr).PartitionExpr()
```

Comment:
I don't think it's a good style, IMO a better way to do it is something like:

```
pt := t.(table.PartitionedTable); err != nil {
    panic(...)
}
pe, err := t.PartitionExpr()
```
Which is more clear. However, some extra changes are required and it might not be a good idea to make it in this PR.

Reply:
However, it is already confirmed that it is a partitioned table, so theoretically this(`t.(partitionExpr)`) should not report an error.